### PR TITLE
Modernize packaging, require pytest 6.2, prepare for pytest 7, other changes

### PR DESCRIPTION
**Pull request by bluetech**
Before we release pytest 7 I figured that pytest-xdist should be ready for it.

Please see the commits for all of the changes. I can break it down to multiple PRs if preferred.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,5 +1,75 @@
...
+
+[options.entry_points]
+pytest11 =
+    xdist = xdist.plugin
+    xdist.looponfail = xdist.looponfail
+
+[options.extras_require]
+testing =
+    filelock
+    pytest>=6.2.0
```
Comment (nicoddemus) [inline setup.cfg:56]: I don't think we need to duplicate this do we?

File: setup.cfg [hunk 2]
```diff
@@ -1,5 +1,75 @@
...
+psutil = psutil>=3.0
+setproctitle = setproctitle
+
 [flake8]
 max-line-length = 100
+
+[mypy]
+mypy_path = src
+# check_untyped_defs = True
```
Comment (nicoddemus) [inline setup.cfg:65]: I think we should remove the commented options, or leave a comment explaining why they should be left uncommented.

File: tox.ini [hunk 3]
```diff
@@ -30,6 +29,14 @@ deps = pytest
 commands =
   pytest {posargs}
 
+[testenv:linting]
```
Comment (nicoddemus) [inline tox.ini:32]: This was returned for local usage, right?

**Review by nicoddemus — APPROVED** [review]
Looks great, thanks a lot for working on this.

Left a few minor comments.

File: tox.ini (hunk 3 above)
Comment (bluetech) [reply tox.ini:32]: Right, I find it convenient to have a tox env for pre-commit as I usually don't have it installed globally.

**Comment by bluetech** [comment]
Thanks for the review @nicoddemus I fixed your comments.

**Pull request merged**